src/detect.c

```diff
@@ -1050,6 +1050,24 @@ DetectRunTxSortHelper(const void *a, const void *b)
 #define TRACE_SID_TXS(sid,txs,...)
 #endif
 
+// Get inner transaction for engine
+void *DetectGetInnerTx(void *tx_ptr, AppProto alproto, AppProto engine_alproto, uint8_t flow_flags)
+{
+    if (alproto == ALPROTO_DOH2) {
```

Comment:
is this expected to be the common case for calls to this function? Or should we put the non-doh2 first to improve branch prediction?